File: 1. Two Sum/TwoSum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> map = new Dictionary<int, int>();
        for (int i = 0, j = nums.Length; i != j; ++i) {
            int difference = target - nums[i];
            if (map.ContainsKey(difference)) {
                return new int[2]{ map[difference], i };
            } else if (!map.ContainsKey(nums[i])) {
                map.Add(nums[i], i);
            }
        }
        return null;
    }
}

File: 1. Two Sum/TwoSum.rs
struct Solution {
}

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        use std::collections::HashMap;
        let mut map : HashMap<i32, usize> = HashMap::with_capacity(nums.len());
        for i in 0..nums.len() {
            let difference = target - nums[i];
            if map.contains_key(&difference) {
                return vec![*map.get(&difference).unwrap() as i32, i as i32];
            }
            map.insert(nums[i], i);
        }
        return vec![];
    }
}

fn main() {
    let ret = Solution::two_sum(vec![2, 7, 11, 15], 9);
    println!("{} {}", ret[0], ret[1]);
    let pat = 
}
